jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/MetaInfConfiguration.java

```diff
@@ -699,38 +635,40 @@ protected List<Resource> findWebInfLibJars(WebAppContext context)
             return List.of();
 
         Resource webInf = context.getWebInf();
-        if (Resources.isReadableDirectory(webInf))
-        {
-            Resource webInfLib = webInf.resolve("lib");
+        if (webInf == null || !webInf.exists() || !webInf.isDirectory())
```

Comment:
I think you should keep the `isReadableDirectory()` check instead of exploding/slightly changing it.